graph_box_umbrella/apps/web_client/src/components/RadarCharts/RadarChart/index.js

```diff
@@ -0,0 +1,212 @@
+      }],
+      options: {
+        chart: {
+          height: 350,
+          type: 'radar',
+          dropShadow: {
+            enabled: true,
+            blur: 1,
+            left: 1,
+            top: 1
+          }
+        },
+        title: {
+          text: 'Radar Chart - Multi Series'
+        },
+        stroke: {
+          width: 2
+        },
+        fill: {
+          opacity: 0.1
+        },
+        markers: {
+          size: 0
+        },
+        xaxis: {
+          categories: ['Category 1', 'Category 2', 'Category 3', 'Category 4', 'Category 5', 'Category 6']
+        },
+      },
+    };
+  }
+
+  pushField() {
+    let newIndex = this.state.series.length + 1;
+    let newData = [0]
+    let newSeriesArray = [...this.state.series, { name: `Series ${newIndex}`, data: [...newData] }];
+    this.setState({
+      series: newSeriesArray
+    });
+  }
+
+  popField() {
+    let newSeriesArray = [...this.state.series];
+    newSeriesArray.pop();
+    this.setState({
+      series: newSeriesArray
+    });
+  }
+
+  pushCategory() {
+    let i = this.state.options.xaxis.categories.length + 1;
+    let newCategories = [...this.state.options.xaxis.categories];
+    newCategories.push(`Category ${i}`);
+    this.setState({
+      options: {
+        xaxis: {
+          categories: [...newCategories]
+        }
+      }
+    });
+  }
+
+  popCategory() {
+    let newCategories = [...this.state.options.xaxis.categories];
+    newCategories.pop();
+    this.setState({
+      options: {
+        xaxis: {
+          categories: [...newCategories]
+        }
+      }
+    });
+  }
+
+  updateFieldName(e, index) {
+    let newSeriesArray = [...this.state.series];
+    newSeriesArray[index] = {...newSeriesArray[index], name:e.target.value};
+    this.setState({
+      series: newSeriesArray
+    });
+  }
+
+  updateCategoryName(e, index) {
+    let newCategories = [...this.state.options.xaxis.categories];
+    newCategories[index] = e.target.value;
+    this.setState({
+      options: {
+        xaxis: {
+          categories: [...newCategories]
+        }
+      }
+    });
+  }
+
+  getMaxSize(seriesArray) {
+    let max = 0;
+    for(let i = 0; i < seriesArray.length; i++) {
+      max = max < seriesArray[i].data.length ? seriesArray[i].data.length : max;
+    }
+    return max;
+  }
+
+  updateFieldData(data, index) {
+    let newData = data.trim().split(' ');
+    let newCategories = [];
+    let previousCategories = [...this.state.options.xaxis.categories];
+    let maxLength = this.getMaxSize([...this.state.series]);
+    for (let i = 0; i < maxLength; i++) {
+      if (previousCategories[i]) {
+        newCategories.push(previousCategories[i]);
+      }
+      else {
+        newCategories.push(`Category ${i+1}`);
+      }
+    }
+    let newSeriesArray = [...this.state.series];
+    newSeriesArray[index] = {...newSeriesArray[index], data: newData};
+    this.setState({
+      series: newSeriesArray,
+      options: {
```

Comment:
I assume you'd want to spread options back into options like 
```Javascript
      series: newSeriesArray,
      options: {
        ...this.state.options,
        xaxis: {
          categories: [...newCategories]
        }
      }
```

otherwise I think you'll overwrite everything in options with `xaxis`